### PR TITLE
Show no records found for report export on mobile for AB#15048

### PR DESCRIPTION
**Pull request by BrianMaki**
# Fixes [AB#15048](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15048)

## Description

For report export show 'No records found' when in mobile view.

Note: Immunization shows 'No records found' when both Immunization and Recommendations have no rows returned.

**Health Visits:**

Screenshot 2023-03-03 at 3 30 24 PM

**Special Authority Requests:**

Screenshot 2023-03-03 at 3 29 44 PM

**Medications:**

Screenshot 2023-03-03 at 3 29 08 PM

**Laboratory Tests:**

Screenshot 2023-03-03 at 3 28 29 PM

**Immunizations:**

**Immunization no rows and recommendations no rows**

Screenshot 2023-03-06 at 1 13 53 PM

Screenshot 2023-03-06 at 1 14 04 PM

**Immunization with rows and recommendations with no rows**

Screenshot 2023-03-06 at 2 30 42 PM

Screenshot 2023-03-06 at 2 29 42 PM

**Immunization no rows and recommendations with rows**

Screenshot 2023-03-06 at 2 36 12 PM

Screenshot 2023-03-06 at 2 36 05 PM

**Hospital Visits:**

Screenshot 2023-03-03 at 3 27 07 PM

**Covid19 Test Results:**

Screenshot 2023-03-03 at 3 25 56 PM

**My Notes:**

Screenshot 2023-03-03 at 3 22 58 PM

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

Yes

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by MikeLyttle — APPROVED** [review]

**Comment by ninosamson** [comment]
@BrianMaki I think we should show "No records found." text only on immunization report if there is no history AND no recommendations. 

Otherwise, the report should be available, download button enabled, and no text displayed.

**Comment by BrianMaki** [comment]
> @BrianMaki I think we should show "No records found." text only on immunization report if there is no history AND no recommendations.
> 
> Otherwise, the report should be available, download button enabled, and no text displayed.

Updated PR.  See screenshots.  @ninosamson let's discuss more during Monday standup.

**Comment by BrianMaki** [comment]
> > @BrianMaki I think we should show "No records found." text only on immunization report if there is no history AND no recommendations.
> > Otherwise, the report should be available, download button enabled, and no text displayed.
> 
> Updated PR. See screenshots. @ninosamson let's discuss more during Monday standup.

@ninosamson 'No records found' is returned when both immunizations and recommendations return no rows.

File: Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue [hunk 1]
```diff
@@ -300,12 +306,12 @@ export default class ImmunizationHistoryReportComponent extends Vue {
             </b-table>
             <b-row class="mt-3">
```
Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue:307]: The `d-none d-md-block` classes could have been applied to this level, every element within is expected to behave the same way.

image

File: Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue [hunk 2]
```diff
@@ -258,12 +258,18 @@ export default class ImmunizationHistoryReportComponent extends Vue {
...
                     <h4>Immunization History</h4>
                 </b-col>
             </b-row>
-            <b-row v-if="isEmpty && !isLoading">
+            <b-row v-if="isEmpty && !isLoading" class="d-none d-md-block">
+                <b-col>No records found.</b-col>
+            </b-row>
+            <b-row
+                v-if="isRecommendationEmpty && isEmpty && !isLoading"
+                class="mt-2 d-sm-block d-md-none"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue:271]: `d-sm-block` won't cover the xs breakpoint; you would need `d-block` for that.  However, neither are needed here as the default for `<b-row>` (display: flex) is sufficient for the element to be rendered.

Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue]: Fixed

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/ReportsComponent.vue [hunk 3]
```diff
@@ -566,7 +566,7 @@ export default class ReportsComponent extends Vue {
         <div
             v-if="reportComponentName"
             data-testid="reportSample"
-            class="sample d-none d-md-block"
+            class="preview"
```
Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/report/ReportsComponent.vue:569]: Would it not have been quicker to use the `@on-is-empty-changed` event to determine if a "no records available"  message could be shown?

Immunization could have watched for records and recommendation changes before raising the event to the parent.

**Review by BronzBierd — COMMENTED** [review]
It appears to be working. I have asked a few questions on the approach.

**Pull request merged**